pkg/codegen/schema/schema.go

```diff
@@ -1235,6 +1280,10 @@ func bindTypes(pkg *Package, complexTypes map[string]ComplexTypeSpec) (*types, e
 			// object type is its token. While this doesn't affect object types directly, it breaks the interning of types
 			// that reference object types (e.g. arrays, maps, unions)
 			typs.objects[token] = &ObjectType{Token: token}
+
+			if strings.Contains(token, "#/resources") {
+				typs.resources[token] = &ResourceType{Token: token}
+			}
```

Comment:
I think that this is dead code: the token for an object type will never contain `#/resources`. What I think you need to do is add a second loop either before or after this loop like so:

```go
for token := range pkg.Resources {
    typs.resources[token] = &ResourceType{Token: token}
}
```